OctopusDeploy/Calamari
# Add GCP authentication support for K8s steps

# Background
This PR updates the `Calamari` component for https://github.com/OctopusDeploy/OctopusDeploy/pull/9242
Changes in this PR consist of:
- Added implementation for K8s context set up using GoogleCloudAccount, including context that uses VM's service account.
- Added unit test for K8s context set up using GoogleCloudAccount.
- Added unit test for K8s Live Fixture which tests against a live K8s cluster hosted in `octopus-api-tester` GCP project.
- Added method to download `gcloud` tool for K8s Live Fixture.

# Testing strategies
This change is tested by two unit tests implemented, including a test against a live K8s instance hosted in GCP.
Furthermore, those smoke tests have been done locally:
- Create and perform health check a deployment target using a Google cloud account.
- Create and perform health check a deployment target using a Google cloud account with impersonation.
- Create and perform health check a deployment target using a VM's service account of a GCP VM hosted in `octopus-api-tester` project.
- Create and perform health check a deployment target using a VM's service account with impersonation.
- Deploy an image using each of the versions of deployment targets created above.
